### PR TITLE
`glibc`: Some bug fixes, plus arc and csky start files

### LINKED ISSUE #14118: s390x: use of undeclared identifier 'newfstatat'
### Zig Version

0.10.0

### Steps to Reproduce and Observed Behavior

```bash
$ zig cc foo.c -target s390x-linux-gnu -mcpu=z10
warning: FileNotFound: /opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/crti.S
warning: FileNotFound: /opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/crtn.S
warning: FileNotFound: /opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/start-2.33.S
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysv/linux/fstatat64.c:51:11: warning: call to undeclared function 'INTERNAL_SYSCALL'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  int r = INTERNAL_SYSCALL_CALL (statx, fd, file, AT_NO_AUTOMOUNT | flag,
          ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:58:3: note: expanded from macro 'INTERNAL_SYSCALL_CALL'
  __INTERNAL_SYSCALL_DISP (__INTERNAL_SYSCALL, __VA_ARGS__)
  ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:52:3: note: expanded from macro '__INTERNAL_SYSCALL_DISP'
  __SYSCALL_CONCAT (b,__INTERNAL_SYSCALL_NARGS(__VA_ARGS__))(__VA_ARGS__)
  ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:28:37: note: expanded from macro '__SYSCALL_CONCAT'
#define __SYSCALL_CONCAT(a,b)       __SYSCALL_CONCAT_X (a, b)
                                    ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:27:37: note: expanded from macro '__SYSCALL_CONCAT_X'
#define __SYSCALL_CONCAT_X(a,b)     a##b
                                    ^
<scratch space>:95:1: note: expanded from here
__INTERNAL_SYSCALL5
^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:42:3: note: expanded from macro '__INTERNAL_SYSCALL5'
  INTERNAL_SYSCALL (name, 5, a1, a2, a3, a4, a5)
  ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysv/linux/fstatat64.c:99:7: warning: call to undeclared function 'INTERNAL_SYSCALL'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  r = INTERNAL_SYSCALL_CALL (newfstatat, fd, file, buf, flag);
      ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:58:3: note: expanded from macro 'INTERNAL_SYSCALL_CALL'
  __INTERNAL_SYSCALL_DISP (__INTERNAL_SYSCALL, __VA_ARGS__)
  ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:52:3: note: expanded from macro '__INTERNAL_SYSCALL_DISP'
  __SYSCALL_CONCAT (b,__INTERNAL_SYSCALL_NARGS(__VA_ARGS__))(__VA_ARGS__)
  ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:28:37: note: expanded from macro '__SYSCALL_CONCAT'
#define __SYSCALL_CONCAT(a,b)       __SYSCALL_CONCAT_X (a, b)
                                    ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:27:37: note: expanded from macro '__SYSCALL_CONCAT_X'
#define __SYSCALL_CONCAT_X(a,b)     a##b
                                    ^
<scratch space>:96:1: note: expanded from here
__INTERNAL_SYSCALL4
^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysdep.h:40:3: note: expanded from macro '__INTERNAL_SYSCALL4'
  INTERNAL_SYSCALL (name, 4, a1, a2, a3, a4)
  ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/glibc/sysdeps/unix/sysv/linux/fstatat64.c:99:30: error: use of undeclared identifier 'newfstatat'
  r = INTERNAL_SYSCALL_CALL (newfstatat, fd, file, buf, flag);
                             ^
2 warnings and 1 error generated.
```

### Expected Behavior

Compile succeed

### LINKED ISSUE #20376: Building C code for `powerpc-linux-gnueabi(hf)` fails due to duplicate `lgammal` symbol
### Zig Version

0.13.0

### Steps to Reproduce and Observed Behavior

```console
❯ cat test.c
#include <stdio.h>
int main(){printf("hello\n");}
❯ zig cc test.c -target powerpc-linux-gnueabi
<unknown>:0: error: symbol 'lgammal' is already defined
```

Also happens for `powerpc-linux-gnueabihf`, but not `powerpc-linux-musl`.

### Expected Behavior

No error.

### LINKED ISSUE #21076: `zig cc` unable to build hello-world for target s390x-linux-gnu
### Zig Version

0.14.0-dev.1052+d6f997259

### Steps to Reproduce and Observed Behavior

```
$ cat test.c
#include <stdio.h>

int main() {
    printf("Hello world!");
}
$ ./zig-linux-x86_64-0.14.0-dev.1052+d6f997259/zig cc -target s390x-linux-gnu.2.19 test.c -o test
error: sub-compilation of glibc crti.o failed
    note: unable to check cache: stat file 'zig-linux-x86_64-0.14.0-dev.1052+d6f997259/lib/libc/glibc/sysdeps/crti.S' failed: FileNotFound
error: unable to build glibc CRT file: SubCompilationFailed
error: sub-compilation of glibc crtn.o failed
    note: unable to check cache: stat file 'zig-linux-x86_64-0.14.0-dev.1052+d6f997259/lib/libc/glibc/sysdeps/crtn.S' failed: FileNotFound
error: sub-compilation of glibc Scrt1.o failed
    note: unable to check cache: stat file 'zig-linux-x86_64-0.14.0-dev.1052+d6f997259/lib/libc/glibc/sysdeps/start-2.33.S' failed: FileNotFound
$ ./zig-linux-x86_64-0.14.0-dev.1052+d6f997259/zig cc -target s390x-linux-gnu test.c -o test
/home/brettjia/.cache/zig/o/b6158625386ad44943fac58bc9e70c2f/m.s:4025:1: error: symbol 'lgammal' is already defined
lgammal:
^
```

### Expected Behavior

Compilation succeeds.

### PR DESCRIPTION
**Pull request by alexrp**
* Fix the `-frounding-math` warning for real this time.
* Fix the `lgammal` duplicate symbol issue.
* Add arc and csky start files for glibc 2.40 and <= 2.33.
* Set asm and include paths for all targets supported by `std.zig.target`.

Closes #14118.
Closes #20376.
Closes #21076.

### CODE REVIEW

**Comment by alexrp** [comment]
See also https://github.com/ziglang/glibc-abi-tool/pull/10 (used that to debug the duplicate symbol issue).

File: src/glibc.zig [hunk 1]
```diff
@@ -809,6 +849,23 @@ pub fn buildSharedObjects(comp: *Compilation, prog_node: std.Progress.Node) !voi
         var opt_symbol_name: ?[]const u8 = null;
         var versions_buffer: [32]u8 = undefined;
         var versions_len: usize = undefined;
+
+        // There can be situations where there are multiple inclusions for the same symbol with
+        // overlapping versions. For example:
```
Comment (andrewrk) [inline src/glibc.zig:854]: seems like a problem with the data, not the unpacking logic, right?

**Review by andrewrk — CHANGES_REQUESTED** [review]

File: src/glibc.zig (hunk 1 above)
Comment (alexrp) [reply src/glibc.zig:854]: Er, I definitely should have made this clear in the comment: The two `lgammal` inclusions in question have different target lists.

Here's the relevant `list_symbols.zig` output:

```
 lgammal:
  library: libm.so
  versions: 2.4 2.23
  targets: arm-linux-gnueabi armeb-linux-gnueabi arm-linux-gnueabihf armeb-linux-gnueabihf sparc-linux-gnu powerpc-linux-gnueabi powerpc-linux-gnueabihf powerpc64-linux-gnu s390x-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.32
  targets: arc-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.29
  targets: csky-linux-gnueabi csky-linux-gnueabihf
 lgammal:
  library: libm.so
  versions: 2.0 2.23
  targets: m68k-linux-gnu mipsel-linux-gnueabihf mips-linux-gnueabihf mipsel-linux-gnueabi mips-linux-gnueabi x86-linux-gnu sparc-linux-gnu powerpc-linux-gnueabi powerpc-linux-gnueabihf mips64el-linux-gnuabi64 mips64-linux-gnuabi64 mips64el-linux-gnuabin32 mips64-linux-gnuabin32
 lgammal:
  library: libm.so
  versions: 2.33
  targets: riscv32-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.17 2.23
  targets: powerpc64le-linux-gnu aarch64-linux-gnu aarch64_be-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.3
  targets: powerpc64-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.2.5 2.23
  targets: x86_64-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.16 2.23
  targets: x86_64-linux-gnux32
 lgammal:
  library: libm.so
  versions: 2.27
  targets: riscv64-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.2 2.23
  targets: sparc64-linux-gnu s390x-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.36
  targets: loongarch64-linux-gnu
```

Comment (alexrp) [reply src/glibc.zig:854]: So, based on my understanding of [the intent here](https://github.com/ziglang/glibc-abi-tool?tab=readme-ov-file#inclusions), I *think* the data is fine, probably?

If you agree, I'll update the comment to be a bit clearer on this.

Comment (andrewrk) [reply src/glibc.zig]: Thanks, this makes sense now.

**Pull request merged**